net/dns/resolver/forwarder.go

```diff
@@ -210,6 +218,57 @@ func (f *forwarder) packetListener(ip netaddr.IP) (packetListener, error) {
 	return lc, nil
 }
 
+func (f *forwarder) getDoHClient(ip netaddr.IP) (urlBase string, c *http.Client, ok bool) {
+	urlBase, ok = knownDoH[ip]
+	if !ok {
+		return
+	}
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if c, ok := f.dohClient[ip]; ok {
+		return urlBase, c, true
+	}
+	if f.dohClient == nil {
+		f.dohClient = map[netaddr.IP]*http.Client{}
+	}
+	nsDialer := netns.NewDialer()
+	c = &http.Client{
+		Transport: &http.Transport{
+			IdleConnTimeout: 30 * time.Second,
```

Comment:
Motivation? Feels short.

Reply:
happy to hear principled alternatives :)

Moved it to a const with docs and declaration of arbitrariness.

Reply:
My instinct says: I ~never change DNS servers. I don’t resolve very often, but when I do I want the latency to be low. Which suggests: no timeout.

Reply:
Added named const & docs.